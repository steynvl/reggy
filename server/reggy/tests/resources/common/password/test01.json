{
  "type": "Password",
  "information": {
    "shouldStartWith": "Anything",
    "shouldContain": [
      "Digit",
      "Special character"
    ],
    "minimumLength": "4",
    "maximumLength": "9"
  },
  "generalRegexInfo": {
    "startRegexMatchAt": "Anywhere",
    "endRegexMatchAt": "Anywhere",
    "regexTarget": "Java"
  },
  "generateMethod": "commonUseCases"
}
