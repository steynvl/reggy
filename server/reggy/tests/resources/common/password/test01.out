Java
Pattern regex = Pattern.compile("(?=.*\\d)(?=.*[!\"#$%&'()*+,-.:;<=>?@\\[\\]^_`{|}~]).{4,9}");
(?=.*\\d)(?=.*[!\"#$%&'()*+,-.:;<=>?@\\[\\]^_`{|}~]).{4,9}
Perl
my $regex = /(?=.*\d)(?=.*[!"#$%&'()*+,-.:;<=>?@\[\]^_`{|}~]).{4,9}/;
(?=.*\d)(?=.*[!"#$%&'()*+,-.:;<=>?@\[\]^_`{|}~]).{4,9}
POSIX
(?=.*\d)(?=.*[!"#$%&'()*+,-.:;<=>?@\[\]^_`{|}~]).{4,9}
(?=.*\d)(?=.*[!"#$%&'()*+,-.:;<=>?@\[\]^_`{|}~]).{4,9}
Python
regex = re.compile(r'(?=.*\d)(?=.*[!"#$%&\'()*+,-.:;<=>?@\[\]^_`{|}~]).{4,9}')
(?=.*\d)(?=.*[!"#$%&\'()*+,-.:;<=>?@\[\]^_`{|}~]).{4,9}
JavaScript
const regex = /(?=.*\d)(?=.*[!"#$%&'()*+,-.:;<=>?@\[\]^_`{|}~]).{4,9}/;
(?=.*\d)(?=.*[!"#$%&'()*+,-.:;<=>?@\[\]^_`{|}~]).{4,9}
PHP
$regex = '/(?=.*\d)(?=.*[!"#$%&'()*+,-.:;<=>?@\[\]^_`{|}~]).{4,9}/';
(?=.*\d)(?=.*[!"#$%&'()*+,-.:;<=>?@\[\]^_`{|}~]).{4,9}
Golang
regex, _ := regexp.Compile("(?=.*\\d)(?=.*[!\"#$%&'()*+,-.:;<=>?@\\[\\]^_`{|}~]).{4,9}")
(?=.*\\d)(?=.*[!\"#$%&'()*+,-.:;<=>?@\\[\\]^_`{|}~]).{4,9}
Rust
let re = Regex::new(r"(?=.*\d)(?=.*[!"#$%&'()*+,-.:;<=>?@\[\]^_`{|}~]).{4,9}").unwrap();
(?=.*\d)(?=.*[!"#$%&'()*+,-.:;<=>?@\[\]^_`{|}~]).{4,9}
C#
var re = new Regex(@"(?=.*\d)(?=.*[!"#$%&'()*+,-.:;<=>?@\[\]^_`{|}~]).{4,9}");
(?=.*\d)(?=.*[!"#$%&'()*+,-.:;<=>?@\[\]^_`{|}~]).{4,9}
Scala
val re = "(?=.*\\d)(?=.*[!\"#$%&'()*+,-.:;<=>?@\\[\\]^_`{|}~]).{4,9}".r
(?=.*\\d)(?=.*[!\"#$%&'()*+,-.:;<=>?@\\[\\]^_`{|}~]).{4,9}
Kotlin
val regex = Regex("(?=.*\\d)(?=.*[!\"#$%&'()*+,-.:;<=>?@\\[\\]^_`{|}~]).{4,9}")
(?=.*\\d)(?=.*[!\"#$%&'()*+,-.:;<=>?@\\[\\]^_`{|}~]).{4,9}